source/macos/AdaptiveCards/AdaptiveCards/Renderers/ColumnSetRenderer.swift

```diff
@@ -61,7 +65,14 @@ class ColumnSetRenderer: BaseCardElementRendererProtocol {
                 columnViews[index].widthAnchor.constraint(equalTo: firstColumn.widthAnchor).isActive = true
             }
             columnSetView.distribution = .fill
-        } else if numberOfAutoItems == totalColumns {
+        } else if numberOfAutoItems > 0 || numberOfAutoItems == totalColumns {
+            let width = ( Constants.maxCardWidth - ( Constants.padding * ( columnViews.count + 1 ))) / columnViews.count
+            
+            for index in (0 ..< columnViews.count) {
+                let widthAnchor = columnViews[index].widthAnchor.constraint(equalToConstant: CGFloat(width))
+                widthAnchor.priority = .defaultHigh
+                widthAnchor.isActive = true
```

Comment:
Adding width anchor will cause problems if those elements are part of toggle visibility right?

Reply:
have checked with toggle visibility case. added gif for same.